botocore/utils.py

```diff
@@ -3315,3 +3315,77 @@ def _serialize_if_needed(self, value, iso=False):
                 return value.isoformat()
             return value.strftime('%Y-%m-%dT%H:%M:%S%Z')
         return value
+
+
+SERVICE_NAME_ALIASES = {'runtime.sagemaker': 'sagemaker-runtime'}
+
+
+def handle_service_name_alias(service_name, **kwargs):
+    return SERVICE_NAME_ALIASES.get(service_name, service_name)
+
+
+# Mapping to determine the service ID for services that do not use it as the
+# model data directory name. The keys are the data directory name and the
+# values are the service IDs. This parameter is considered private and is
+# subject to abrupt breaking changes or removal without prior announcement.
+_CLIENT_NAME_TO_HYPHENIZED_SERVICE_ID_OVERRIDES = {
```

Comment:
Can we instead just document that this map is private as part of the comment and remove the leading underscore? It may also help making the note that this is private as the very first thing said in the comment so it is obvious and maybe change the wording from it being "private" to it being "not a public interface" to clarify that "we" can use this map but people consuming botocore should not.

In general, we should not be marking this private with an underscore as that would indicate that it can only be used within this classes/functions of this `utils.py` file, which is not the case. We use the map in the config provider.